runtime/parachains/src/configuration.rs

```diff
@@ -361,9 +360,22 @@ pub mod pallet {
 		StorageValue<_, HostConfiguration<T::BlockNumber>, ValueQuery>;
 
 	/// Pending configuration (if any) for the next session.
+	///
+	/// DEPRECATED: This is no longer used, and will be removed in the future.
 	#[pallet::storage]
 	pub(crate) type PendingConfig<T: Config> =
-		StorageMap<_, Twox64Concat, SessionIndex, HostConfiguration<T::BlockNumber>>;
+		StorageMap<_, Twox64Concat, SessionIndex, migration::v1::HostConfiguration<T::BlockNumber>>;
+
+	/// Pending configuration changes.
+	///
+	/// This is a list of configuration changes, each with a session index at which it should
+	/// be applied.
+	///
+	/// The list is sorted ascending by session index. Also, this list can only contain at most
+	/// 2 items: for the next session and for the `scheduled_session`.
+	#[pallet::storage]
+	pub(crate) type PendingConfigs<T: Config> =
+		StorageValue<_, Vec<(SessionIndex, HostConfiguration<T::BlockNumber>)>, ValueQuery>;
```

Comment:
if it can only contain 2 items, why not use a fixed size array?

Reply:
What would that give though? At the same time, it would be more annoying to change if we were to change the SESSION_DELAY.

Reply:
No specific reason, just thinking that if you are going to have some specific constraint about this storage item, that we can just program that in.

Reply:
Yeah, I am generally pro making it more typesafe, but I believe here it does not provide much benefit while can be potential problem in the future (although I admit that it is also not super likely we will change SESSION_DELAY)